charts/csm-authorization/charts/redis/templates/redis.yaml

```diff
@@ -1,87 +1,150 @@
-apiVersion: apps/v1 
-kind: Deployment
+apiVersion: v1
+kind: Service
 metadata:
-  name: redis-primary
+  name: {{ .Values.redis.name }}
   namespace: {{ include "custom.namespace" . }}
-  labels:
-    app: redis
 spec:
+  type:
+  clusterIP: None
+  selector:
+    app: {{ .Values.redis.name }}
+  ports:
+  - protocol: TCP
+    port: 6379
+    targetPort: 6379
+    name: {{ .Values.redis.name }}
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Values.redis.name }}
+  namespace: {{ include "custom.namespace" . }}
+spec:
+  serviceName: {{ .Values.redis.name }}
+  replicas: {{ .Values.redis.replicas }}
   selector:
     matchLabels:
-      app: redis
-      role: primary
-      tier: backend
-  replicas: 1
+      app: {{ .Values.redis.name }}
   template:
     metadata:
       labels:
-        app: redis
-        role: primary
-        tier: backend
+        app: {{ .Values.redis.name }}
+      annotations:
+        checksum/secret: {{ include (print $.Template.BasePath "/redis-secret.yaml") . | sha256sum }}
     spec:
+      initContainers:
+      - name: config
+        image: {{ .Values.redis.images.redis }}
+        env:
+          - name: REDIS_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: redis-csm-secret
+                key: password
+
+        command: [ "sh", "-c" ]
+        args:
+          - |
+            cp /csm-auth-redis-cm/redis.conf /etc/redis/redis.conf
+            echo "masterauth $REDIS_PASSWORD" >> /etc/redis/redis.conf
+            echo "requirepass $REDIS_PASSWORD" >> /etc/redis/redis.conf
+            
+            echo "Finding master..."
```

Comment:
master/slave terminology will be flagged by other code checks. Update to different terms where possible.